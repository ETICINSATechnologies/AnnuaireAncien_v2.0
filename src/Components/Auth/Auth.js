class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
