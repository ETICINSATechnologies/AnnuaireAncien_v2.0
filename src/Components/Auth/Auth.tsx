class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
