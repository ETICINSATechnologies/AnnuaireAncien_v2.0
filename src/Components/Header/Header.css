.Header {
  background-image: var(--backgroundHeader);
  vertical-align: middle;
  border-bottom: 0.3vmin solid black;
  width: 100%;
  height: 100%;
}

.Header .header_a {
  display: grid;
  grid-template-columns: 12vw 1fr 12vw;

  vertical-align: middle;
  text-decoration: none;

  border-bottom: 0.3vmin solid black;

  width: 100%;
  height: 100%;
}

.Header .home_logo {
  grid-row: 1;
  grid-column: 1;
  margin: auto 1vw;
  padding: 0;
  font-size: 2vw;
  height: 80%;
}

.Header .title {
  color: var(--titleHeader);
  text-align: center;
  margin: auto;
  text-transform: uppercase;
  font-size: 5vw;
  width: 100%;
  max-height: 100%;
}
