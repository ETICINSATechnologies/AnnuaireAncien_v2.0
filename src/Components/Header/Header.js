import React, {Component} from 'react';
import './Header.css';

import home_logo from '../../Images/home_logo.png'
import {Link} from "react-router-dom";

class Header extends Component {
    render() {
        return (
            <header className='Header'>
                <Link className="header_a" to="/">
                    <img className="home_logo" src={home_logo} alt=""/>
                    <h1 className="title"> Annuaire des Anciens </h1>
                </Link>
            </header>

        );
    }
}

export default Header;
