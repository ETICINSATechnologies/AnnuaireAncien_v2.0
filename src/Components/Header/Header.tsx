import React, { Component } from "react";
import "./Header.css";
import { Link, Redirect } from "react-router-dom";

interface Props {
  logo: string;
}
const Header = (props: Props) => {
  return (
    <header className="Header">
      <Link className="header_a" to="/">
        <img className="home_logo" src={require("../../" + props.logo)} />
        <h1 className="title"> Annuaire des Anciens </h1>
      </Link>
    </header>
  );
};

export default Header;
