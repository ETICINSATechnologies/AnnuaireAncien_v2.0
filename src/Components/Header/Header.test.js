import React from 'react';
import Header from './Header';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from '../../Components/Auth/Auth';

Enzyme.configure({adapter: new Adapter()});

describe('Header component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<Header/>).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('Header unit test', () => {
        expect(Enzyme.shallow(<Header/>)
            .find('.Header')
            .length
        ).toBe(1)
    });

});