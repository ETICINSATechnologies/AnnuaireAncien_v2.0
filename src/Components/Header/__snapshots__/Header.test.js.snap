// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without crashing 1`] = `
<header
  className="Header"
>
  <a
    className="header_a"
    href="/"
  >
    <img
      alt=""
      className="home_logo"
      src="home_logo.png"
    />
    <h1
      className="title"
    >
       Annuaire des Anciens
    </h1>
  </a>
</header>
`;
